src/psyclone/domain/lfric/algorithm/lfric_alg.py

```diff
@@ -129,7 +130,7 @@ def create_from_kernel(self, name, kernel_path):
         # arbitrary value, we use an *integer* literal for this, irrespective
         # of the actual type of the scalar argument. The compiler/run-time will
         # take care of appropriate type casting.
-        psyir.add_lfric_precision_symbol(table, "i_def")
+        table.add_lfric_precision_symbol("i_def")
```

Comment:
I am so glad you agreed. I have to admit that I was aware that the outcome of the previous telco was to use stand-alone functions, but ... tbh, that was so wrong for me, I thought I just try it (since it wouldn't be too hard to refactor this). I am glad that you are happy with this (and not upset ... but ... some things just have to be done in a certain way, and ... of course I am very objective here :grin:)

Reply:
I think if we have something where we're consistently passing in an object which will be modified then it should be a method of that object. Hopefully no right-thinking developer would disagree with that? ;-)